client/src/app/controllers/search-project-controller/search-project.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AuthService } from '../auth-controller/auth.service';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchProjectService {
+  apiUrl = environment.apiUrl;
+
+  constructor(private http: HttpClient, private authService: AuthService) { }
+  //This method is pretty simple, just fill in the fields that exist or will be used with the search or null if not. To the best of my knowledge, js/ts does 
+  //not support named parameters so if you wanted to only serach with deadline, the correct call to this method would be searchProjectsMultipleParams(null, null, null, null, null, null, DATE PARAMETER)
+  searchProjectsMultipleParams(query: string | null, majors: string[] | null, GPA: number | null, npp: number | null,
```

Comment:
You should DEFINITELY replace this with a custom interface that contains these properties. It could go in the src/app/_models directory and look something like this:
```
export interface SearchOptions {
  query?: string,
  majors?: string[],
  GPA?: number,
  npp?: number,
  pageNum?: number,
  posted?: Date,
  deadline?: Date,
};
```
The question marks mark that the field is optional.

Reply:
ok